src/serializer.js

```diff
@@ -16,7 +16,28 @@ var Serializer = DS.RESTSerializer.extend({
   },
   extractDeleteRecords: function(store, type, payload) {
     return this.extractArray(store, type, payload);
+  },
+  serialize: function(snapshot, options) {
+    var hash = this._super(snapshot, options);
+    return this.pickQueriedFields(hash, snapshot, options);
+  },
+  pickQueriedFields: function(data, record) {
```

Comment:
method accepts 2 args but you pass 3 https://github.com/collectrium/ember-data-socketio-adapter/pull/37/files#diff-4568d5ad520ddb6de4962d49a9de1047R22